apps/aechannel/src/aesc_fsm.erl

```diff
@@ -3184,8 +3215,19 @@ watch_req() ->
 on_chain_id(#data{on_chain_id = ID} = D, _) when ID =/= undefined ->
     {ID, D};
 on_chain_id(D, SignedTx) ->
-    {ok, PubKey} = aesc_utils:channel_pubkey(SignedTx),
-    {PubKey, D#data{on_chain_id = PubKey}}.
+    try aesc_utils:channel_pubkey(SignedTx) of
+        {ok, PubKey} ->
+            {PubKey, D#data{on_chain_id = PubKey}}
+    ?CATCH_LOG(_E)
+        if D#data.strict_checks == false ->
```

Comment:
`strict_checks` is for AliceFSM, in the case Alice tries cheating Bob. That indeed would be a test scenario. I am wondering if we get here at all.

Reply:
Yep, the `check_incorrect_create` goes right there, this is indeed needed